.github/workflows/ci.yml

```diff
@@ -98,7 +98,7 @@ jobs:
       - name: ruff
         if: matrix.run_lint == true
         run: |
-          ruff .
+          ruff check .
```

Comment:
This a mixture of CI and version bumps, as written earlier:
- CI changes are in seperate PR
- version bumps are in a seperate PR.

This is done for several reasons:
- mixed changes are harder to bisect
- version bumps need to be clearly identifiable in the changelog for users
- CI changes and version bumps needs extra handling in github actions

Reply:
Split the CI change out to #2242.

Reply:
This one still contains the CI part.

Reply:
Removed it from this one(although tests fail again without that).

Reply:
The test should not be affected by the 3rd party version numbers, but I will ask CI to run on this PR (due to CI setup it is more than just starting CI).